src/client/cli/cmd/disable_zones.cpp

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "disable_zones.h"
+
+#include "animated_spinner.h"
+#include "common_callbacks.h"
+#include "common_cli.h"
+#include "multipass/cli/argparser.h"
+#include "multipass/cli/client_common.h"
+
+#include <QCommandLineOption>
+
+namespace multipass::cmd
+{
+ReturnCode DisableZones::run(ArgParser* parser)
+{
+    if (const auto ret = parse_args(parser); ret != ParseCode::Ok)
+        return parser->returnCodeFrom(ret);
+
+    if (ask_for_confirmation)
+    {
+        const PlainPrompter prompter{term};
+        auto answer = prompter.prompt("This operation will forcefully stop the VMs in zone3. Proceed? (Yes/No)");
+        while (!std::regex_match(answer, client::yes_answer) && !std::regex_match(answer, client::no_answer))
+            answer = prompter.prompt("Please answer (Yes/No)");
+
+        if (std::regex_match(answer, client::no_answer))
+            return Ok;
+    }
+
+    AnimatedSpinner spinner{cout};
+    spinner.start(format("Disabling {}", fmt::join(zone_names, ", ")));
+
+    ZonesStateRequest request{};
+    for (const auto& zone_name : zone_names)
+        request.add_zones(zone_name);
+    request.set_available(false);
+    request.set_verbosity_level(parser->verbosityLevel());
+
+    const auto on_success = [&](const ZonesStateReply&) {
+        spinner.stop();
+        const auto n_zones = zone_names.size();
+        cout << n_zones << (n_zones == 1 ? " zone" : " zones") << " disabled successfully" << std::endl;
```

Comment:
I tried a few commands to see what they say on success:

1. `start`, `stop`, `delete`, `recover` or `purge`, `set`, `mount`, `umount` don't print anything.
2. `launch` prints `Launched: <instance>`. 
3. `snapshot` says `Snapshot taken: <instance>.<snapshot>`.
4. `restore` prints `Snapshot restored: <instance>.<snapshot>`

In 2 and 3, the success message is used to inform the user of the name that was chosen. 4 falls a little off of this scheme. 

I don't mind being a little more informative. It may come in handy if we ever implement wildcards. But could we stick to the same format, i.e. something like `[Zone[s]] Disabled: <zones>`?